src/lib/Form/Type/UserRegisterType.php

```diff
@@ -38,9 +48,26 @@ public function getParent()
 
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        $allowedFieldsId = $this
+            ->configResolver
+            ->getParameter('user_registration.form.allowed_field_definitions_identifiers');
```

Comment:
This is part of the event listener logic. I would move it to the listener itself (which also makes it a little more self-contained).